Unix/base/oiomi.h

```diff
@@ -53,6 +53,156 @@ FILE_EVENT0(10005, trace_EngineProcessTerminated_Impl, LOG_CRIT, PAL_T("abnormal
+#if defined(CONFIG_ENABLE_DEBUG)
+#define trace_Failed_Socket_Directory_Create(a0) trace_Failed_Socket_Directory_Create_Impl(__FILE__, __LINE__, scs(a0))
+#else
+#define trace_Failed_Socket_Directory_Create(a0) trace_Failed_Socket_Directory_Create_Impl(0, 0, scs(a0))
+#endif
+FILE_EVENT1(10024, trace_Failed_Socket_Directory_Create_Impl, LOG_CRIT, PAL_T("Failed to create socket directory: %s"), const char *)
+#if defined(CONFIG_ENABLE_DEBUG)
+#define trace_Failed_Socket_Directory_Chown(a0) trace_Failed_Socket_Directory_Chown_Impl(__FILE__, __LINE__, scs(a0))
+#else
+#define trace_Failed_Socket_Directory_Chown(a0) trace_Failed_Socket_Directory_Chown_Impl(0, 0, scs(a0))
+#endif
+FILE_EVENT1(10025, trace_Failed_Socket_Directory_Chown_Impl, LOG_CRIT, PAL_T("Failed to chown socket directory: %s"), const char *)
+#if defined(CONFIG_ENABLE_DEBUG)
+#define trace_Failed_Generate_Socket_File_Name() trace_Failed_Generate_Socket_File_Name_Impl(__FILE__, __LINE__)
+#else
+#define trace_Failed_Generate_Socket_File_Name() trace_Failed_Generate_Socket_File_Name_Impl(0, 0)
+#endif
+FILE_EVENT0(10026, trace_Failed_Generate_Socket_File_Name_Impl, LOG_CRIT, PAL_T("Failed to generate socket file name"))
+#if defined(CONFIG_ENABLE_DEBUG)
+#define trace_Failed_Generate_Secret_String() trace_Failed_Generate_Secret_String_Impl(__FILE__, __LINE__)
+#else
+#define trace_Failed_Generate_Secret_String() trace_Failed_Generate_Secret_String_Impl(0, 0)
+#endif
+FILE_EVENT0(10027, trace_Failed_Generate_Secret_String_Impl, LOG_CRIT, PAL_T("Failed to generate secret string"))
+#if defined(CONFIG_ENABLE_DEBUG)
+#define trace_Failed_Socket_Pair() trace_Failed_Socket_Pair_Impl(__FILE__, __LINE__)
+#else
+#define trace_Failed_Socket_Pair() trace_Failed_Socket_Pair_Impl(0, 0)
+#endif
+FILE_EVENT0(10028, trace_Failed_Socket_Pair_Impl, LOG_CRIT, PAL_T("Failed to create socket pair"))
+#if defined(CONFIG_ENABLE_DEBUG)
+#define trace_Failed_Fork() trace_Failed_Fork_Impl(__FILE__, __LINE__)
+#else
+#define trace_Failed_Fork() trace_Failed_Fork_Impl(0, 0)
+#endif
+FILE_EVENT0(10029, trace_Failed_Fork_Impl, LOG_CRIT, PAL_T("Failed to fork"))
+#if defined(CONFIG_ENABLE_DEBUG)
+#define trace_Failed_Set_User_Engine() trace_Failed_Set_User_Engine_Impl(__FILE__, __LINE__)
+#else
+#define trace_Failed_Set_User_Engine() trace_Failed_Set_User_Engine_Impl(0, 0)
+#endif
+FILE_EVENT0(10030, trace_Failed_Set_User_Engine_Impl, LOG_CRIT, PAL_T("Failed to set uid/gid of engine"))
+#if defined(CONFIG_ENABLE_DEBUG)
+#define trace_Failed_Launch_Engine() trace_Failed_Launch_Engine_Impl(__FILE__, __LINE__)
+#else
+#define trace_Failed_Launch_Engine() trace_Failed_Launch_Engine_Impl(0, 0)
+#endif
+FILE_EVENT0(10031, trace_Failed_Launch_Engine_Impl, LOG_CRIT, PAL_T("Failed to launch engine"))
+#if defined(CONFIG_ENABLE_DEBUG)
 #define trace_AgentMgrHandleRequestForNS(a0) trace_AgentMgrHandleRequestForNS_Impl(__FILE__, __LINE__, tcs(a0))
```

Comment:
It's odd that the next message isn't an error, which is what's in the source file. What will this do for our ability to upgrade while maintaining message IDs?

Reply:
I saw LOG_CRIT is 10000-10006, LOG_ERR is 20000-20148, LOG_WARNING is 30000-30213, LOG_INFO is 40000-40043, LOG_DEBUG is 45000-45380, LOG_VERBOSE is 55000-55056 and 60000-60024,
then a 60000-60017 for LOG_DEBUG at the end.
Duplicated ids are from 60000-60017 now, they used different FILE_EVENT_number function.

Reply:
This was an error on my part; at quick glance, I hadn't realized that the GitHub UI was not showing me contiguous lines.

Reply:
I am moving Bruce's auth-related trace messages from 60000-block to 50000-block.  This avoids conflict, but breaks backward compatibility a bit.  Can't do anything about that.